Jenkinsfile.github

```diff
@@ -44,6 +45,9 @@ pipeline {
     stages {
         stage('Build Prep') {
             steps {
+                script {
+                    jenkinsUtils.writeNetRc()
```

Comment:
What does this do?

Reply:
This is a cherry-pick of a commit Eli made to add authentication when accessing Python modules. Here's his commit message:

```
CRAYSAT-1620: Use authentication when accessing csm-python-modules

This commit adds creation of a .netrc file on the Jenkins host on which the
build is running. The intention of this file is to provide credential
authentication in the case when the package's dependencies are being installed
in a VirtualEnv on the host to facilitate running unit tests. This uses the
utility jenkinsUtils.writeNetRc() found in csm-shared-library.
```